light-client/relayer/src/relayer.rs

```diff
@@ -0,0 +1,163 @@
+        logger: Logger,
+    ) -> Self {
+        let next_block_index = config.start_block_index;
+        Self {
+            config,
+            next_block_index,
+            ledger_db,
+            watcher_db,
+            sender,
+            logger,
+        }
+    }
+
+    /// Entrypoint for relayer loop.
+    pub fn entry_point(&mut self) -> ! {
+        // Poll ledger for data to relay
+        loop {
+            match self.ledger_db.get_block_data(self.next_block_index) {
+                Err(LedgerError::NotFound) => std::thread::sleep(Self::POLLING_FREQUENCY),
+                Err(e) => {
+                    log::error!(
+                        self.logger,
+                        "Unexpected error when checking for block data {}: {:?}",
+                        self.next_block_index,
+                        e
+                    );
+                    std::thread::sleep(Self::ERROR_RETRY_FREQUENCY);
+                }
+                Ok(block_data) => {
+                    if let Err(err) = self.process_block(&block_data) {
+                        log::error!(
+                            self.logger,
+                            "When processing block {}: {:?}",
+                            self.next_block_index,
+                            err
+                        );
+                        std::thread::sleep(Self::ERROR_RETRY_FREQUENCY);
+                    } else {
+                        self.next_block_index += 1;
+                    }
+                }
+            }
+        }
+    }
+
+    fn process_block(&mut self, block_data: &BlockData) -> Result<(), WatcherDBError> {
+        // Tracing
+        let tracer = tracer!();
+
+        let mut span =
+            block_span_builder(&tracer, "poll_block", self.next_block_index).start(&tracer);
+
+        span.set_attribute(TELEMETRY_BLOCK_INDEX_KEY.i64(self.next_block_index as i64));
+
+        // First, check if there's anything interesting in this block
+        let relevant_burns = Self::check_for_relevant_burns(&block_data.contents().outputs);
+
+        // There are no burns with relevant memos in this block so this block is
+        // finished processing.
+        if relevant_burns.is_empty() {
+            return Ok(());
+        }
+
+        // Try to get signatures from the watcher
+        // Loop until we get enough signatures
+        tracer.in_span("loop_for_signatures", |_cx| {
+            loop {
+                let signatures = self.check_for_signatures(self.next_block_index)?;
+                if signatures.len() >= self.config.min_signatures {
+                    self.sender.send(
+                        relevant_burns,
+                        block_data.block(),
+                        block_data.contents(),
+                        signatures,
+                    );
+                    return Ok(());
+                } else {
+                    // We didn't get enough signatures, but let's assume that more are coming soon
```

Comment:
:+1: I like the "let's assume that more are coming" it helps communicate why this code is looping and polling instead of erroring.